libnopegl/src/backends/vk/bindgroup_vk.c

```diff
@@ -289,7 +290,14 @@ int ngli_bindgroup_vk_update_texture(struct bindgroup *s, int32_t index, const s
     struct gpu_ctx_vk *gpu_ctx_vk = (struct gpu_ctx_vk *)s->gpu_ctx;
 
     struct texture_binding_vk *texture_binding = ngli_darray_get(&s_priv->texture_bindings, index);
-    texture_binding->texture = binding->texture ? binding->texture : gpu_ctx_vk->dummy_texture;
+
+    NGLI_RC_UNREFP(&texture_binding->texture);
```

Comment:
Is it possible that we could have the same texture passed down the update_texture? In which case an unref here would destroy the texture and thus the assignment below would be void? Maybe overkill but maybe we should unref it after the new one has been referenced?

Reply:
That can't happen as the caller must own a reference to the texture when calling update_texture().
IE:
```
    struct texture *tex = ngli_texture_create();
    ngli_bindgroup_update_texture(b, tex);
    ngli_bindgroup_update_texture(b, tex);
```
The only way to trigger the scenario you described is to somehow call ngli_texture_freep(tex) and still call update_texture() with that dangling pointer.